src/compiler/utilities.ts

```diff
@@ -1926,27 +1926,55 @@ namespace ts {
         return node;
     }
 
-    export function isSourceTreeNode(node: Node): boolean {
+    /**
+     * Gets a value indicating whether a node originated in the parse tree.
+     *
+     * @param node The node to test.
+     */
+    export function isParseTreeNode(node: Node): boolean {
         return (node.flags & NodeFlags.Synthesized) === 0;
     }
 
-    export function getSourceTreeNode(node: Node): Node {
-        if (isSourceTreeNode(node)) {
+    /**
+     * Gets the original parse tree node for a node.
+     *
+     * @param node The original node.
+     * @returns The original parse tree node if found; otherwise, undefined.
+     */
+    export function getParseTreeNode(node: Node): Node;
+
+    /**
+     * Gets the original parse tree node for a node.
+     *
+     * @param node The original node.
+     * @param nodeTest A callback used to ensure the correct type of parse tree node is returned.
+     * @returns The original parse tree node if found; otherwise, undefined.
+     */
+    export function getParseTreeNode<T extends Node>(node: Node, nodeTest?: (node: Node) => node is T): T;
+    export function getParseTreeNode(node: Node, nodeTest?: (node: Node) => boolean): Node {
+        if (isParseTreeNode(node)) {
             return node;
         }
 
         node = getOriginalNode(node);
 
-        if (isSourceTreeNode(node)) {
+        if (isParseTreeNode(node) && (!nodeTest || nodeTest(node))) {
             return node;
         }
 
         return undefined;
     }
 
-    export function getSourceTreeNodeOfType<T extends Node>(node: T, nodeTest: (node: Node) => node is T): T {
-        const source = getSourceTreeNode(node);
-        return source && nodeTest(source) ? source : undefined;
+    export function getOriginalSourceFiles(sourceFiles: SourceFile[]) {

```

Comment:
It's used for emitting declaration files: https://github.com/Microsoft/TypeScript/blob/transforms-extractFromPrinter/src/compiler/printer.ts#L197